src/mmw/js/src/core/views.js

```diff
@@ -265,11 +265,17 @@ var MapView = Marionette.ItemView.extend({
                 maximumAge : maxAge,
                 timeout : timeout
             };
-            navigator.geolocation.getCurrentPosition(
-                geolocation_success,
-                _.noop,
-                geolocationOptions
-            );
+
+            // Wait a bit and then get the location. Suppresses issues on Safari
+            // in which immediately requesting the location results in an
+            // infinite request loop for geolocation permissions.
+            setTimeout(function() {
+                navigator.geolocation.getCurrentPosition(
+                    geolocation_success,
+                    _.noop,
+                    geolocationOptions
+                );
+            }, 500);
```

Comment:
Since this is an arbitrary number, consider calling it out as `GEOLOCATION_DELAY`or something where we define other defaults: https://github.com/WikiWatershed/model-my-watershed/blob/develop/src/mmw/js/src/core/views.js#L189-L191